### ISSUE TITLE
Debug mode active after migration 9.4.5 to 9.5.1

### ISSUE THREAD

**Issue by carlosbmc** [glpi-project/glpi#7848]
After migration debug mode is enabled,  when try to disable it then returrns this error

Uncaught Exception SodiumException: public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes in /glpi/inc/toolbox.class.php at line 293

**Comment by fralla2** [comment]
Same problem here migrating from the same version...
Just after the install, I sign-in to the login page and receive this error : 

![image](https://user-images.githubusercontent.com/38400135/90553204-febb8180-e161-11ea-8075-eab8c261477e.png)

Thanks

**Comment by fralla2** [comment]
php-errors.log : 

`[2020-08-18 15:12:55] glpiphplog.CRITICAL:   *** Uncaught Exception SodiumException: public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes in /var/www/html/glpi-9.5.1/inc/toolbox.class.php at line 293
  Backtrace :
  inc/toolbox.class.php:293                          sodium_crypto_aead_xchacha20poly1305_ietf_decrypt()
  inc/authldap.class.php:2677                        Toolbox::sodiumDecrypt()
  inc/auth.class.php:225                             AuthLDAP::tryToConnectToServer()
  inc/authldap.class.php:2797                        Auth->connection_ldap()
  inc/authldap.class.php:2852                        AuthLDAP::ldapAuth()
  inc/auth.class.php:843                             AuthLDAP::tryLdapAuth()
  front/login.php:84                                 Auth->login()
`

**Comment by carlosbmc** [comment]
Disabling SODIUM in PHP server solves the problem

**Comment by fralla2** [comment]
Thanks for the tip @carlosbmc 

Unfortunately for me it didn't work as once this module disabled, LDAP auth now throws this error message :

`[2020-08-18 15:33:14] glpiphplog.ERROR: Toolbox::sodiumDecrypt() in /var/www/html/glpi-9.5.1/inc/toolbox.class.php line 296
Unable to decrypt content. It may have been crypted with another key.  {"user":"@glpi05"}
[2020-08-18 15:33:14] glpiphplog.ERROR: Auth::connection_ldap() in /var/www/html/glpi-9.5.1/inc/auth.class.php line 251
Something went wrong searching in LDAP directory  {"user":"@glpi05","mem_usage":"0.004\", 3.11Mio)"}`

**Comment by carlosbmc** [comment]
After disabling SODIUM you need to rewrite LDAP password

**Comment by cedric-anne** [comment]
Hi @carlosbmc @fralla2 ,

Can you tell us which version of libsodium and/or sodim PHP extension is installed on your system ?

Regards

**Comment by fralla2** [comment]
`# php --ri sodium
sodium support => enabled
libsodium headers version => 1.0.18
libsodium library version => 1.0.18`


`# php --version
PHP 7.2.32 (cli) (built: Jul  8 2020 07:33:50) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.32, Copyright (c) 1999-2018, by Zend Technologies`

**Comment by fralla2** [comment]
Note : Updating GLPI from 9.4.5 to 9.5.0 did not throw that error but 9.4.5 to 9.5.1 is...

Thanks
Francois

**Comment by cedric-anne** [comment]
What is the size of your `config/glpicrypt.key` file ? It should be 32 bytes length.

**Comment by cedric-anne** [comment]
I investigate on this and it seems to be due to some data that were not migrated correctly.

#7782 should prenvent this kind of case.
#7869 would prevent the exception.

**Comment by fralla2** [comment]
> Which result did you get when checking file length with ls -l ?

Good question, I tried to recreate the issue since I had deleted the file and tested the new fix but this time the file was of the correct size but if I manually recreate the file uging the content previously noted, it gives me a file of 75.

`)3lØ<95>¸v·éþ^@^Lw<83>»ó<86><8c>ý\s8<9a>è><8a><8e><89>y<8f><92>Á`

Note : I tested fix #7875 and the migration worked perfectly.

Thanks
Francois

**Comment by cedric-anne** [comment]
Hi,

> but if I manually recreate the file uging the content previously noted, it gives me a file of 75.
I guess contents use chars that cannot be printed/copied/pasted with default file editor encoding. Manipulating them is not something that should be done, so this is not an issue.

Many fixes have been done on security key management, are you still facing issues or can we close this one ?

Regards

**Comment by fralla2** [comment]
For me, pull #7782 , #7869 and #7875  have fixed the issue.

I think we can close it except if @carlosbmc still have problem?

Thanks
Francois

**Comment by cedric-anne** [comment]
Hi,

Then I close the issue. Feel free to reopen if the problem appears again.

Regards

**Comment by anyqov** [comment]
If I have already migrated from 9.4.6 to 9.5.1 and get the error "public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes", what do I need to do to fix it? There have already been many changes in the database after the migration and I would like to save them.

**Comment by carlosbmc** [comment]
Change PHP to last version



Enviado desde mi smartphone Samsung Galaxy.



-------- Mensaje original --------
De: anyqov <notifications@github.com>
Fecha: 7/10/20 0:24 (GMT+01:00)
Para: glpi-project/glpi <glpi@noreply.github.com>
Cc: carlosbmc <carlosbmc@hotmail.com>, Mention <mention@noreply.github.com>
Asunto: Re: [glpi-project/glpi] Debug mode active after migration 9.4.5 to 9.5.1 (#7848)


If I have already migrated from 9.4.6 to 9.5.1 and get the error "public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes", what do I need to do to fix it? There have already been many changes in the database after the migration and I would like to save them.

**Comment by anyqov** [comment]
Error: "public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes".
I got this problem when I updated GLPI from 9.4.4 to 9.5.1.
Versions: PHP 7.4.11 (latest), 10.4.14-MariaDB, Centos 7.

```
$ sudo php /var/www/glpi/bin/console glpi:security:change_key
Found 2 field(s) and 3 configuration entries requiring migration.
Do you want to continue ? [Yes/no]yes

In toolbox.class.php line 293:

  public nonce size should be SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES bytes

glpi:security:change_key
```

How do I set it up?

**Comment by cedric-anne** [comment]
Hi,

Many fixes have beend done on this part in GLPI 9.5.2. Please update GLPI and try again.

Regards
